### PR TITLE
Bump dependencies old Ruby versions

### PR DESCRIPTION
**Pull request by JuanVqz**
Description:

continue updating dependencies and making them work in the GitHub Actions

If changes to the behavior are made, clearly describe what changes.

I will abide by the [code of conduct](https://github.com/fastruby/skunk/blob/main/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: skunk.gemspec [hunk 1]
```diff
@@ -39,7 +39,6 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
   spec.add_dependency "rubycritic", ">= 4.5.2", "< 5.0"
-  spec.add_dependency "terminal-table", "~> 3.0"
```
Comment (arielj) [inline skunk.gemspec:42]: I think we have to keep the dependencies in this file or it won't be installed when installing the gem (when installing the gem I understand it doesn't check the Gemfile) and they won't show up in rubygems

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ on:
 
 jobs:
   test-ruby-2-4-x:
-    runs-on: ubuntu-latest
```
Comment (arielj) [inline .github/workflows/main.yml:14]: why is this change needed?

Comment (JuanVqz) [reply .github/workflows/main.yml:14]: isn't [this](https://github.com/fastruby/skunk/actions/runs/4845701187/jobs/8634780556?pr=109) the issue related to the ubuntu version?

File: skunk.gemspec (hunk 1 above)
Comment (JuanVqz) [reply skunk.gemspec:42]: I'll check it then

Comment (etagwerker) [reply skunk.gemspec:42]: yeah, good point, we do require `terminal-table` as a dependency because it's used to generate the console report. So removing it from here will cause problems...

Comment (JuanVqz) [reply skunk.gemspec]: Good catch! Moved back to the skunk.gemspec file

File: skunk.gemspec [hunk 3]
```diff
@@ -48,8 +48,6 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "minitest-stub_any_instance", "~> 1.0.2"
   spec.add_development_dependency "minitest-stub-const", "~> 0.6"
   spec.add_development_dependency "rake", "~> 13.0"
-  spec.add_development_dependency "reek", "~> 6.1"
-  spec.add_development_dependency "rubocop", "~> 1.48"
```
Comment (arielj) [inline skunk.gemspec:52]: I would still list these 2 here (without a version restriction), so they show up as dev dependencies in rubygems, and then you keep the version definition in the specific Gemfiles

**Review by etagwerker — APPROVED** [review]
@JuanVqz Looks good, could you please add a line to the CHANGELOG file?

**Review by etagwerker — APPROVED** [review]
